### PR TITLE
[front] feat: swipe up to refresh videos in comparison page

### PR DESCRIPTION
**Pull request by GresilleSiffle**
**related issues** #1931

---

### Description

In the comparison page, this PR allows the users to trigger the Auto button by swiping up on the entity card of an entity selector. For now, swiping down doesn't do anything.

**Additional changes**

The UI is slightly more compact, to make the submit button more visible. The gap between the weekly collective goal component and the entity selectors has been reduced on mobile. The extra gap created by the entity contexts has been removed when there is no context to display.

**to-do**

*swipe ux*
- [x] clearly delimit the swipe-able surface
- [x] set a comfortable minimum velocity for the swipe (instead of the default 0.5)

*swipe logic*
- [x] disable swipe in `EntitySelectorInnerAuth` when `variant === "noControl"`

**improvements**
- [x] do not display the entity contexts container when there is no context
- add the swipe down interaction, allowing to go back in the history
- [x] avoid hiding the criteria container when an entity is loading (simply disable the submit button)

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
  - if not, delete the related issues section
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [ ] The tests pass and have been updated if relevant
- [x] The code quality check pass

[development-guidelines]: https://github.com/tournesol-app/tournesol/blob/main/CONTRIBUTING.md#development-guidelines

### CODE REVIEW

File: frontend/src/features/entity_selector/EntitySelector.tsx [hunk 1]
```diff
@@ -267,12 +330,15 @@ const EntitySelectorInnerAuth = ({
...
+                htmlId={`auto-suggestion-${alignment}`}
                 disabled={loading}
                 currentUid={uid}
                 otherUid={otherUid}
-                onClick={() => {
+                onClick={async () => {
                   setLoading(true);
                   setInputValue('');
+                  setSlideIn(false);
+                  await wait(ENTITY_CARD_SWIPE_TIMEOUT + 8);
```
Comment (GresilleSiffle) [inline frontend/src/features/entity_selector/EntitySelector.tsx:341]: The `wait` function ensures the swipe up animation is displayed before displaying a new video, but using `setTimeout` might not be very elegant. What do you think?

An alternative could be to move the logic of the `askNewVideo` callback from the component `AutoEntityButton` to `EntitySelector`. This way `setTimeout` won't be required anymore as `EntitySelector` will be able to easily trigger the video refresh after the swipe up animation. Therefore, the component `AutoEntityButton` will be a simple button that requires its parent to implement the refresh logic.

File: frontend/src/features/entity_selector/EntitySelector.tsx [hunk 2]
```diff
@@ -283,102 +349,114 @@ const EntitySelectorInnerAuth = ({
...
+        mountOnEnter
+        in={slideIn}
+        direction={slideDirection}
+        onEntered={onSlideEntered}
+        onExited={onSlideExited}
+        timeout={ENTITY_CARD_SWIPE_TIMEOUT}
+        appear={variant === 'noControl' ? false : true}
+      >
+        {/* position: relative is required to correctly display the entity unavailable box */}
+        <Box {...bindDrag()} sx={{ touchAction: 'pan-x' }} position="relative">
```
Comment (amatissart) [inline frontend/src/features/entity_selector/EntitySelector.tsx:362]: For some reason, `pan-x` does not behave as expected on iOS 17.4 and cause all vertical gestures to be ignored:
```suggestion
        <Box {...bindDrag()} sx={{ touchAction: 'none' }} position="relative">
```

File: frontend/src/features/goals/CollectiveGoalWeeklyProgress.tsx [hunk 3]
```diff
@@ -31,7 +33,7 @@ const CollectiveGoalWeeklyProgress = () => {
       flexDirection="column"
       alignItems="center"
       gap={1}
-      mb={4}
+      mb={smallScreen ? 2 : 4}
```
[the remark this answers is not in the document]
Comment (GresilleSiffle) [reply frontend/src/features/goals/CollectiveGoalWeeklyProgress.tsx]: Fixed by https://github.com/tournesol-app/tournesol/pull/1930/commits/76bb8a7ce2ea1892037d10145cfd9c70e79f917e

File: frontend/src/features/entity_selector/EntitySelector.tsx [hunk 4]
```diff
@@ -257,6 +319,7 @@ const EntitySelectorInnerAuth = ({
             spacing={1}
             display={uid ? 'flex' : 'none'}
             direction={alignment === 'left' ? 'row' : 'row-reverse'}
+            justifyContent={smallScreen ? 'space-around' : 'flex-start'}
```
[the remark this answers is not in the document]
Comment (GresilleSiffle) [reply frontend/src/features/entity_selector/EntitySelector.tsx]: Fixed by https://github.com/tournesol-app/tournesol/pull/1930/commits/76bb8a7ce2ea1892037d10145cfd9c70e79f917e

**Pull request merged**